CRM/Member/BAO/Membership.php

```diff
@@ -2505,13 +2492,17 @@ public static function recordMembershipContribution(&$params, $ids = array()) {
     // store contribution id
     $params['contribution_id'] = $contribution->id;
 
-    //insert payment record for this membership
-    if (empty($ids['contribution']) || !empty($params['is_recur'])) {
-      CRM_Member_BAO_MembershipPayment::create(array(
+    // Create membership payment if it does not already exist
+    $membershipPayment = civicrm_api3('MembershipPayment', 'get', [
+      'contribution_id' => $contribution->id,
+    ]);
+    if (empty($membershipPayment['count'])) {
```

Comment:
OK so thinking through this - it used to say 'if this is not an update to an existing contribution or it is updating a recurring one then create a membership payment record'. Now it says if no membership payment record exists then create one. I'm not quite sure the implications, if any, of this change but I'd rather it wasn't in the same PR as tidy ups to which function is called because it feels like it 'could' have unintended consequences and I don't know how solid test cover is here

Reply:
Ok, I've put that in a separate PR https://github.com/civicrm/civicrm-core/pull/14897  This one should be ok to merge now?